### DISCUSSION TITLE
How to resolve and remove "Warning: Content: Support for defaultProps..."

### DISCUSSION THREAD

**Discussion by jeremycaine** [vercel/next.js#52522] [Help] [answered]
### Summary

Bootstrapped a simple Next 13 app and attempting to add in IBM Carbon components. 

Following the advice in [docs](https://nextjs.org/docs/getting-started/react-essentials) and use of `use client`

But I get the warning ...
```
Warning: Content: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.
    at Content (webpack-internal:///(sc_client)/./node_modules/@carbon/react/lib/components/UIShell/Content.js:25:16)
```

I looked for similar issues in other frameworks - like this Material UI on [https://github.com/robcaldecott/nextjs-mui-app-folder](https://github.com/robcaldecott/nextjs-mui-app-folder)

I can see maybe there are fixes in MUI components that are not in IBM Carbon - but architecturally I think my simple app is structured the same as this MUI example - I did a JS version of it in parallel to mine [here in this subfolder](https://github.com/jeremycaine/next13-with-carbon/tree/main/alt-mui) - and it did not have the same Warning.

### Example

https://github.com/jeremycaine/next13-with-carbon

**Comment by icyJoseph** [comment]
Hi,

The UIShell->Content Component from IBC Carbon does use defaultProps.

Added to https://twitter.com/AndaristRake/status/1597514454920724480, in particular: https://github.com/facebook/react/pull/25699

> React 18.3 will remove default props on function components

This warning makes pretty much sense. I am not sure what are you expecting. 

- Default props usage: https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/UIShell/Content.js#L31
- Open issue with regards to React 18 compatibility: https://github.com/carbon-design-system/carbon/issues/10777

### Edit

- Actual tracker issue from IBM Carbon https://github.com/carbon-design-system/carbon/issues/13424

**Reply by jeremycaine** [reply]
So I get that this warning is from React as to an upcoming change they will make, and also that IBM Carbon is using defaultProps.

I was wondering therefore if the Next 13 guidance on 'use client' etc was in some way going to circumvent this behaviour.

**Reply by icyJoseph** [reply]
Mmm I think we might be mixing up a couple of things here. 

`use client` has nothing to do here, AFAIK. That directive is only to opt-in to the old component behaviour, where there is a JavaScript bundle that hydrates a component/page, etc... the same thing we've been doing up till now.

- https://react.dev/blog/2023/05/03/react-canaries
- https://nextjs.org/blog/next-13-4#what-does-app-router-stability-mean

I wonder if the canary usage is what's making this warning surface. 

The root of the issue is that IBM Carbon ought to update their function components.

**Reply by jeremycaine** [reply]
you have helped me understand more thank you - I need `use client` anyway with carbon since they are client components - but this issue on props is different - but fundamentally, yes - the IBM Carbon team need to update their components.

I did upgrade to canary release the other day - I will drop back to a stable release and see whether the message appears.

**Reply by tay1orjones** [reply, marked as answer]
Hey, IBM Carbon maintainer here 👋 We have an issue tracking the refactor of defaultProps usages to use default parameters instead: https://github.com/carbon-design-system/carbon/issues/13424

**Reply by masoudmhmdi** [reply]
encountered this error when I was using Next.js v14 with the react-mention library:
 "Mention: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead."
 https://github.com/signavio/react-mentions
